pkg/translator/prometheusremotewrite/testutils_test.go

```diff
@@ -212,6 +212,34 @@ func getHistogramDataPointWithExemplars(t *testing.T, time time.Time, value floa
 	return h
 }
 
+// getHistogramDataPointWithExemplarsInt returns a HistogramDataPoint populating the exemplar with an int64 value.
+func getHistogramDataPointWithExemplarsInt(t *testing.T, time time.Time, value int64, traceID string, spanID string, attributeKey string, attributeValue string) pmetric.HistogramDataPoint {
```

Comment:
I basically copy/paste the content of this function ⬇️

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/dd600c17034e1e4cbee80a07274b65777fdf9eda/pkg/translator/prometheusremotewrite/testutils_test.go#L187

But changing the signature to receive an `int64` value instead of a `float64` one. Does this make sense?

I'm also added a comment in the top of the function, but I notice that most of the private functions doesnt have an comment, should I keep it?

Reply:
Maybe this is a good use case for [generics](https://go.dev/doc/tutorial/generics)?

```go
func getHistogramDataPointWithExemplars[V int64 | float64](t *testing.T, time time.Time, value V, traceID string, spanID string, attributeKey string, attributeValue string) pmetric.HistogramDataPoint {
.
.
.
    switch V.(type) {
    case int64:
        e.SetIntValue(value)
    case float64:
        e.SetDoubleValue(value)
    }
.
.
.
}
```

We could avoid a little bit of code and the change is not thaaaat big to the original function

Reply:
[7644383](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36688/commits/76443830b483d3f945cf291faa1e68e2a04ca694)

I removed the new helper function that I created, and I added the `generic` that you mentioned in the existing function